busy_beaver/blueprints/slack/slash_command.py

```diff
@@ -87,58 +92,25 @@ def command_not_found(**data):
     logger.info("[Busy Beaver] Unknown command")
     return make_slack_response(text="Command not found. Try `/busybeaver help`")
 
-
-##########################################
-# Associate GitHub account with Slack user
-# TODO refactor this
-##########################################
 @slash_command_dispatcher.on("connect")
 def link_github(**data):
-    logger.info("[Busy Beaver] New user. Linking GitHub account.")
-    slack_id = data["user_id"]
-    user_record = User.query.filter_by(slack_id=slack_id).first()
-
-    if user_record:
-        logger.info("[Busy Beaver] Slack acount already linked to GitHub")
-        return make_slack_response(text=ACCOUNT_ALREADY_ASSOCIATED)
-
-    user = User()
-    user.slack_id = slack_id
-    user = add_tracking_identifer_and_save_record(user)
-    attachment = create_github_account_attachment(user.github_state)
-    return make_slack_response(text=VERIFY_ACCOUNT, attachments=attachment)
+    logger.info("[Busy Beaver] Slash command to associate with Github account.")
+    message, attachment = generate_account_attachment(**data)
```

Comment:
We are passing in the entire `data` dictionary into the `generate_account_attachment` function. It's a bit cryptic as to what information the function actually needs.

If we extract the params we need from the data dictionary and pass them into the view, things become a bit easier to read.

Reply:
Which of the following ways to clarify the data view do you envision?

1. `message, attachment = generate_account_attachment(data["user_id"]) `
2. `user_id = data["user_id"]
message, attachment = generate_account_attachment(user_id)`